src/fe/inf_fe.C

```diff
@@ -339,6 +322,78 @@ void InfFE<Dim,T_radial,T_map>::reinit(const Elem * inf_elem,
 
 
 
+template <unsigned int Dim, FEFamily T_radial, InfMapType T_map>
+void InfFE<Dim, T_radial, T_map>::determine_calculations()
+{
+  this->calculations_started = true;
+
+  // If the user forgot to request anything, but we're enabling
+  // deprecated backwards compatibility, then we'll be safe and
+  // calculate everything.  If we haven't enable deprecated backwards
+  // compatibility then we'll scream and die.
+#ifdef LIBMESH_ENABLE_DEPRECATED
+  if (!this->calculate_nothing &&
+      !this->calculate_phi && !this->calculate_dphi &&
+      !this->calculate_phi_scaled && !this->calculate_dphi_scaled &&
+      !this->calculate_xyz && !this->calculate_jxw &&
+      !this->calculate_map_scaled && !this->calculate_map &&
+#ifdef LIBMESH_ENABLE_SECOND_DERIVATIVES
+      !this->calculate_d2phi &&
+#endif
+      !this->calculate_curl_phi && !this->calculate_div_phi)
+    {
+      libmesh_deprecated();
+      this->calculate_phi = this->calculate_dphi = this->calculate_jxw = true;
+#ifdef LIBMESH_ENABLE_SECOND_DERIVATIVES
+      this->calculate_d2phi = true;
+#endif
+      this->calculate_phi_scaled = this->calculate_dphi_scaled = this->calculate_xyz = true;
+      if (FEInterface::field_type(fe_type.family) == TYPE_VECTOR)
+        {
+          this->calculate_curl_phi = true;
+          this->calculate_div_phi  = true;
+        }
+    }
+#else //LIBMESH_ENABLE_DEPRECATED
+  libmesh_assert (this->calculate_nothing ||
```

Comment:
Does this chain of boolean operators make sense?  I'm trying to parse it (after a `-Werror -Wparentheses` warning kept me from compiling) and I'm failing.

On a related note, we must need to add `--enable-paranoid-warnings --enable-werror` to our infinite elements recipe in CI?

Reply:
oh, you are right: I think, the `!this->`-terms must have no negation here and the `&&` should be changed to `||` to make actually sense.
I think, the issue here is that one cannot chain-up statements with mixed `&&` and `||`; so it should be fixed by adjusting this?  But I currently don't see why `-Wparentheses` prevents parsing of that statement...

I obviously use `--enable-deprecated`; so I didn't catch this mistake.